static/js/components/UserPagePersonalDialog.js

```diff
@@ -46,10 +50,10 @@ export default class UserPagePersonalDialog extends React.Component {
       </Button>,
       <Button
         type='button'
-        className='primary-button save-button'
+        className={`primary-button save-button ${inFlight ? 'disabled-with-spinner' : ''}`}
         key='save'
-        onClick={this.savePersonalInfo}>
-        Save
+        onClick={inFlight ? undefined : this.savePersonalInfo}>
+        {inFlight ? <Spinner singleColor /> : 'Save'}
```

Comment:
This is outside of the scope of this PR, but we have a lot of repeated code in these action buttons - it might make sense to pull that out as a higher level abstraction, maybe something like this:

```js
const profileDialogActions = (onCancel, onSave, inFlight) => ([
  <Button ... />,
  <Button ... />,
]);
```